IanGRap/Squish
# Lobby multiple game modes 2

Contributors: Alfredo, Coulter, the Squish team for their love and support

It's finally here! After 2 months of blood sweat and programmer art, you too can navigate a half decent menu. 

Changes: 
Scripts
* Team.cs: added a new function for removing players from the Teams list. Used during the Character Select Team menu when a player gets off the platform
* CharacterSelectPodium.cs: used during the the CharacterSelectFreeForAllMenu. Handles the podium game object's interactions with the player
* CharacterSelectTeamPlatforms: used during the CharacterSelectTeamMenu. Handles turning on and off the related game objects for the menu
* RegisterTeamMemeber.cs: used with the CharacterSelectTeamMenu colliders. Detect when a player has chosen a team by standing on one of the podiums
* TeamsReadyUp.cs: used during CharacterSelectTeamsMenu. Keeps track of players from both teams and readys them up when appropriate
* ControllerSignIn.cs: refactored to handle the different inputs needed for different states of the lobby. Calls different functions depending on where in the lobby the player is
* PlayerHatChanger.cs: fixed a bug issue where players tried to fix hats even though they aren't spawned
* ActivePlayers.cs: added a helper function to make it easier to get the PlayerData from gameobject
* Skin.cs: updated a function to swap the player colors when they join a specific team
* LobbyStateManager.cs: updated to handle the new states of the Lobby. 
* PlayerSelectUiManager.cs: updates the way the ready screen is activated
* PlayerSpawner.cs: adds new spawn locations for the free for all and teams menu
* PlayerInput.cs: updates when the player's input should be deactivated
* Materials.cs: swaps between different material lists depending on whether the team or free for all mode is selected
* CInput.cs: adds an extension method for ints that allows us to get the PlayerIndex number
* GameState.cs: merges Ian's changes with my changes for managing states in the game
* SceneTransitioner: keeps a record of what game mode was selected in the lobby
* UiDisplayControls.cs: turns on and off the Ui for the character select part of the menus
* GameModeScriptableObject.cs: used to create scriptable objects for the game modes
* UiHorizontalSelectorScriptableObject.cs: Controls the side scrolling Ui used to select the game mode
* GameStartTimer.cs: controls how the timer works when all the players step on the ready platform
* WinScreenController.cs: changed the method used in the WinScreen for calling a new scene
* 

Lobby.unity - added new Ui and game objects for the menus. Most of it is temporary and needs art.
Updated prefabs
* Player Podium.prefab - used during the PlayerSelectFreeForAllMenu. 
* Character Select Team Platforms.prefab - used during the PlayerSelectTeamMenu. Holds the game objects and colliders for registering the team
* ObjectSpawner.prefab - Updated the scripts to hold more info about where to spawn the player during the menus
* AudioManager.prefab - added the new temp sound for the counter
* ControllerManager.prefab - children updated to have events for when the players despawn
* LobbyStateManager.prefab - added new camera positions for when the menus are activated
* MaterialManager.prefab - added Team Skins for the PlayerSelectTeamMenu
* PlayerSelectUIManager.prefab - Updated the Ready Screen UI to use the new button system
* CharacterSelectUI.prefab - added new UI for the Character Select Free For All and Teams menu
* CounterObject.prefab - Created a Counter/Timer for the Lobby Tutorial area. When the players hop on the ready podium, the timer activates
* GameModeMenu.prefab - updated the Menu where you select the Game Mode. It now uses the screen transitioner
* UIHorizontalMenuScriptableObject - part of the GameModeMenu changes
* TombStoneMainMenu.prefab - prefabbed this to make it easier to move between scenes
* LobbyActiveUiScreenManager.prefab - Updated to set the initial UI object to activate

LobbyGameModeRefactor.unity - This screen held all of the changes for the lobby before merging in to Lobby.unity

ScriptableObjects:
* Classic.asset - holds info for the classic mode
* BlockDunk.asset - holds info for the BlockDunk mode
* TreasureGrab.asset - holds the info for the TreasureGrab mode

RiflePrimary2.ogg - added a temporary sound from opengameart.org to fill in for the timer ending sound. This needs to be replaced

How to Test: 
* Open the Lobby.unity scene. Run through the menu
* You should be able to enter all 3 of the game modes through the menu
* When you select Block Dunk, you should be put in the Team Select Menu vs the Free For all menu
* Make sure you can run through the whole game multiple times by only using the in game options
* Any feedback on the code and menu is appreciated
* Right now all of the art is programmer art

P.S I'm sorry
